internal/provider/job_resource.go

```diff
@@ -0,0 +1,518 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var (
+	_ resource.Resource                = &JobResource{}
+	_ resource.ResourceWithImportState = &JobResource{}
+)
+
+func NewJobResource() resource.Resource {
+	return &JobResource{}
+}
+
+// inventoryDataSource is the data source implementation.
+type JobResource struct {
+	client *AAPClient
+}
+
+// Metadata returns the resource type name.
+func (r *JobResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_job"
+}
+
+var _ validator.String = ansibleVarsValidator{}
+
+// ansibleVarsValidator validates that a string Attribute's is a valid JSON.
+type ansibleVarsValidator struct{}
+
+// Description describes the validation in plain text formatting.
+func (validator ansibleVarsValidator) Description(_ context.Context) string {
+	return "string must be a valid JSON."
+}
+
+// MarkdownDescription describes the validation in Markdown formatting.
+func (validator ansibleVarsValidator) MarkdownDescription(ctx context.Context) string {
+	return validator.Description(ctx)
+}
+
+// Validate performs the validation.
+func (v ansibleVarsValidator) ValidateString(ctx context.Context, request validator.StringRequest, response *validator.StringResponse) {
+	if request.ConfigValue.IsNull() || request.ConfigValue.IsUnknown() {
+		return
+	}
+	if !json.Valid([]byte(request.ConfigValue.ValueString())) {
+		response.Diagnostics.Append(validatordiag.InvalidAttributeValueLengthDiagnostic(
+			request.Path,
+			v.Description(ctx),
+			fmt.Sprintf("Invalid JSON string => [%s]", request.ConfigValue.ValueString()),
+		))
+		return
+	}
+}
+
+func ansible_json_vars_validator() validator.String {
+	return ansibleVarsValidator{}
+}
+
+// Schema defines the schema for the resource.
+func (d *JobResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"job_template_id": schema.Int64Attribute{
+				Required: true,
+			},
+			"inventory_id": schema.Int64Attribute{
+				Optional: true,
+			},
+			"execution_environment_id": schema.Int64Attribute{
+				Optional: true,
+			},
+			"job_type": schema.StringAttribute{
+				Computed: true,
+			},
+			"job_url": schema.StringAttribute{
+				Computed: true,
+			},
+			"status": schema.StringAttribute{
+				Computed: true,
+			},
+			"wait_for_completion": schema.BoolAttribute{
+				Optional: true,
+			},
+			"wait_duration": schema.Int64Attribute{
+				Optional: true,
+			},
+			"extra_vars": schema.StringAttribute{
```

Comment:
I tried to define it as map, but it will be only map of strings knowing that ansible variable can be dict/string/list... This is definitely the recommended way to deal with this attribute as this is threat on the provider side but directly passed to the server. More info [here](https://developer.hashicorp.com/terraform/language/expressions/type-constraints#dynamic-types-the-any-constraint)

Reply:
Got it, makes sense. Thanks!

Reply:
I feel like this would be better off as a custom [normalized jsontype](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-jsontypes@v0.1.0/jsontypes#Normalized.Type). This way we aren't reliant on a user having used `jsonencode()`.

Reply:
updated